### PR TITLE
6735 embedded file extractor retries

### PR DESCRIPTION
**Pull request by rcordovano**

### CODE REVIEW

File: Core/src/org/sleuthkit/autopsy/modules/embeddedfileextractor/Bundle.properties [hunk 1]
```diff
@@ -31,8 +31,6 @@ EmbeddedFileExtractorIngestModule.ImageExtractor.pptxContainer.init.err=Pptx con
 EmbeddedFileExtractorIngestModule.ImageExtractor.xlsContainer.init.err=Xls container could not be initialized while reading: {0}
 EmbeddedFileExtractorIngestModule.ImageExtractor.xlsxContainer.init.err=Xlsx container could not be initialized while reading: {0}
 EmbeddedFileExtractorIngestModule.ImageExtractor.extractImage.addToDB.exception.msg=Unable to add the derived files to the database.
-EmbeddedFileExtractorIngestModule.ImageExtractor.getOutputFolderPath.exception.msg=Could not get path for image extraction from Abstract File: {0}
-EmbeddedFileExtractorIngestModule.ImageExtractor.getOutputFolderPath.exception.msg=Could not get path for image extraction from Abstract File: {0}
```
Comment (rcordovano) [inline Core/src/org/sleuthkit/autopsy/modules/embeddedfileextractor/Bundle.properties:35]: We don't localize log messages.

File: Core/src/org/sleuthkit/autopsy/modules/embeddedfileextractor/Bundle.properties-MERGED [hunk 2]
```diff
@@ -1,5 +1,6 @@
 CannotCreateOutputFolder=Unable to create output folder.
 CannotRunFileTypeDetection=Unable to run file type detection.
+EmbeddedFileExtractor_make_output_dir_err=Failed to create module output directory for Embedded File Extractor
```
Comment (rcordovano) [inline Core/src/org/sleuthkit/autopsy/modules/embeddedfileextractor/Bundle.properties-MERGED:3]: This message is for an ingest module start up exception that is display to the user, so it must be localized.

File: Core/src/org/sleuthkit/autopsy/modules/embeddedfileextractor/Bundle_ja.properties [hunk 3]
```diff
@@ -22,7 +22,6 @@ EmbeddedFileExtractorIngestModule.ArchiveExtractor.unpack.warnMsg.zipBomb=ZIP\u7
 EmbeddedFileExtractorIngestModule.ImageExtractor.docContainer.init.err=\u8aad\u307f\u53d6\u308a\u4e2d\u306bDOC\u30b3\u30f3\u30c6\u30ca\u30fc\u3092\u521d\u671f\u5316\u3067\u304d\u307e\u305b\u3093\u3067\u3057\u305f 
 EmbeddedFileExtractorIngestModule.ImageExtractor.docxContainer.init.err=\u6b21\u3092\u8aad\u307f\u53d6\u308a\u4e2d\u306bDOCX\u30b3\u30f3\u30c6\u30ca\u30fc\u3092\u521d\u671f\u5316\u3067\u304d\u307e\u305b\u3093\u3067\u3057\u305f\: {0}
 EmbeddedFileExtractorIngestModule.ImageExtractor.extractImage.addToDB.exception.msg=\u6d3e\u751f\u30d5\u30a1\u30a4\u30eb\u3092\u30c7\u30fc\u30bf\u30d9\u30fc\u30b9\u306b\u8ffd\u52a0\u3067\u304d\u307e\u305b\u3093.
-EmbeddedFileExtractorIngestModule.ImageExtractor.getOutputFolderPath.exception.msg=\u6b21\u306e\u62bd\u8c61\u30d5\u30a1\u30a4\u30eb\u304b\u3089\u306e\u30a4\u30e1\u30fc\u30b8\u62bd\u51fa\u30d1\u30b9\u3092\u53d6\u5f97\u3067\u304d\u307e\u305b\u3093\u3067\u3057\u305f\: {0}
```
Comment (rcordovano) [inline Core/src/org/sleuthkit/autopsy/modules/embeddedfileextractor/Bundle_ja.properties:25]: This duplicated error message, displayed to the user, was updated and moved into an annotation.

File: Core/src/org/sleuthkit/autopsy/modules/embeddedfileextractor/Bundle.properties-MERGED [hunk 4]
```diff
@@ -43,8 +44,6 @@ EmbeddedFileExtractorIngestModule.ImageExtractor.pptxContainer.init.err=Pptx con
 EmbeddedFileExtractorIngestModule.ImageExtractor.xlsContainer.init.err=Xls container could not be initialized while reading: {0}
 EmbeddedFileExtractorIngestModule.ImageExtractor.xlsxContainer.init.err=Xlsx container could not be initialized while reading: {0}
 EmbeddedFileExtractorIngestModule.ImageExtractor.extractImage.addToDB.exception.msg=Unable to add the derived files to the database.
-EmbeddedFileExtractorIngestModule.ImageExtractor.getOutputFolderPath.exception.msg=Could not get path for image extraction from Abstract File: {0}
-EmbeddedFileExtractorIngestModule.ImageExtractor.getOutputFolderPath.exception.msg=Could not get path for image extraction from Abstract File: {0}
```
Comment (rcordovano) [inline Core/src/org/sleuthkit/autopsy/modules/embeddedfileextractor/Bundle.properties-MERGED:47]: This duplicated error message, displayed to the user, was updated and moved into an annotation.

File: Core/src/org/sleuthkit/autopsy/threadutils/TaskRetryUtil.java [hunk 5]
```diff
@@ -0,0 +1,280 @@
...
+            }
+            this.delay = delay;
+            this.timeOut = timeOut;
+            this.timeUnit = timeUnit;
+        }
+
+        /**
+         * Gets the optional delay before the task should be attempted, may be
+         * zero.
+         *
```
Comment (APriestman) [inline Core/src/org/sleuthkit/autopsy/threadutils/TaskRetryUtil.java:121]: I think I would note here that the units for the delay (and the timeout in the next getter) come from getTimeUnit()

File: Core/src/org/sleuthkit/autopsy/threadutils/TaskRetryUtil.java [hunk 6]
```diff
@@ -0,0 +1,280 @@
...
+        public TaskAttempt(Long delay, TimeUnit timeUnit) {
+            if (delay == null || delay < 0) {
+                throw new IllegalArgumentException(String.format("Argument for delay parameter = %d, must be zero or any positive integer", delay));
+            }
+            if (timeUnit == null) {
+                throw new IllegalArgumentException("Argument for timeUnit parameter is null");
+            }
+            this.delay = delay;
+            this.timeOut = 0L;
+            this.timeUnit = TimeUnit.SECONDS;
```
Comment (APriestman) [inline Core/src/org/sleuthkit/autopsy/threadutils/TaskRetryUtil.java:89]: I think this should be saving the passed in timeUnit

**Review by APriestman — CHANGES_REQUESTED** [review]

File: Core/src/org/sleuthkit/autopsy/threadutils/TaskRetryUtil.java (hunk 6 above)
Comment (rcordovano) [reply Core/src/org/sleuthkit/autopsy/threadutils/TaskRetryUtil.java]: Yes, good catch, thanks.

**Review by APriestman — APPROVED** [review]

**Pull request merged**